packages/ai-ide/src/browser/style/index.css

```diff
@@ -66,84 +155,255 @@
 #ai-variable-configuration-container-widget,
 #ai-agent-configuration-container-widget,
 #ai-model-aliases-configuration-widget {
-  margin-top: 5px;
+  flex: 1;
+  overflow: hidden;
 }
 
-/* Variable Settings */
-#ai-variable-configuration-container-widget ul {
-  list-style: none;
-  padding: 0;
-  margin: 0;
+/* Section headers and spacing */
+.settings-section-title {
+  font-size: var(--theia-ui-font-size3);
+  font-weight: 600;
+  margin: 0 0 calc(var(--theia-ui-padding) * 1.5) 0;
+  padding: 0 0 calc(var(--theia-ui-padding) / 2) 0;
+  border-bottom: var(--theia-border-width) solid var(--theia-widget-border);
+}
+
+.settings-section-title pre {
+  font-size: calc(var(--theia-ui-font-size0) * 0.9);
+  color: var(--theia-descriptionForeground);
+  margin-top: calc(var(--theia-ui-padding) / 2);
+}
+
+/* Agent title with toggle */
+.agent-title-with-toggle {
+  display: block;
 }
 
-#ai-variable-configuration-container-widget .variable-item {
+.agent-title-content {
   display: flex;
-  flex-direction: column;
-  margin-bottom: 1rem;
+  justify-content: space-between;
+  align-items: flex-start;
+  gap: calc(var(--theia-ui-padding) * 2);
+}
+
+.ai-id-label {
+  display: inline-block;
+  margin-left: var(--theia-ui-padding);
 }
 
-#ai-variable-configuration-container-widget .variable-args {
+.agent-enable-toggle {
+  display: flex;
+  align-items: center;
+  gap: calc(var(--theia-ui-padding));
+  cursor: pointer;
+  user-select: none;
+  flex-shrink: 0;
+}
+
+/* Toggle switch */
+.toggle-switch {
+  position: relative;
+  display: inline-block;
+  width: 36px;
+  height: 20px;
+}
+
+.toggle-switch input[type="checkbox"] {
+  opacity: 0;
+  width: 0;
+  height: 0;
+}
+
+.toggle-slider {
+  position: absolute;
+  cursor: pointer;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: var(--theia-input-background);
```

Comment:
This is unfortunately not a good fit, for the theia light theme the background becomes invisible, we could use `--theia-radio-inactiveHoverBackground` instead.

<img width="386" height="50" alt="Image" src="https://github.com/user-attachments/assets/6eab4e83-4f3a-41c5-a991-25de41e19e7a" />